src/js/lib/admin-page/components/WipeMetadataModal.js

```diff
@@ -0,0 +1,40 @@
+import { useState } from '@wordpress/element';
+import { TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const WipeMetadataModal = ( { setting, apikey, onConfirm, modalControl } ) => {
+	const [ message, setMessage ] = useState( '' );
+	const [ flagSet, setFlagSet ] = useState( setting.parsely_wipe_metadata_cache );
+
+	const setFlag = ( val ) => {
+		if ( ! val ) {
+			setMessage( '' );
+			setFlagSet( false );
+			onConfirm( val );
+		}
+		if ( val && message === apikey ) {
+			onConfirm( val );
+		}
+		modalControl( false );
+	};
+
+	return (
+		<div className="modal-container">
+			<div onClick={ () => setFlag( false ) } className="modal-background">
+			</div>
+			<div className="modal">
+				<strong>{ __( 'Type', 'wp-parsely' ) } <span className="code">{ apikey }</span> { __( 'below and then click Confirm if you really want to delete all stored metadata. This action cannot be undone', 'wp-parsely' ) }</strong>
```

Comment:
Does the JS `__()` support placeholders? Right now, this is split into separate strings, which makes it harder for translators:

- `'Type'`
- `'below and then [...]'`

With a placeholder, then it can be a single string:

- `'Type %s below and then [...]'`
...with the `%s` then replaced with `<code>{ apikey }</code>` (not sure why a span with a class name is used instead of a `code` element) after the translation has potentially happened.
...and add a `/* translators: API Key */` translator's comment to tell the translator what the placeholder will contain.